opal-ui/src/components/admin/providers/AddIdentityProviderDialog.vue

```diff
@@ -0,0 +1,348 @@
+                >
+                </q-input>
+                <q-input
+                  v-model.number="newProvider.readTimeout"
+                  dense
+                  type="number"
+                  :label="$t('identity_provider.read_timeout')"
+                  :hint="$t('identity_provider.read_timeout_hint')"
+                  class="q-mb-md"
+                  lazy-rules
+                >
+                </q-input>
+                <q-input
+                  v-model="newProvider.callbackURL"
+                  dense
+                  type="text"
+                  :label="$t('identity_provider.callback_url')"
+                  :hint="$t('identity_provider.callback_url_hint')"
+                  class="q-mb-md"
+                  lazy-rules
+                  :rules="[validateUri('validation.identity_provider.callback_url_format', false)]"
+                >
+                </q-input>
+              </q-expansion-item>
+            </q-list>
+          </div>
+        </q-form>
+      </q-card-section>
+
+      <q-separator />
+
+      <q-card-actions align="right" class="bg-grey-3"
+        ><q-btn flat :label="$t('cancel')" color="secondary" v-close-popup />
+        <q-btn flat :label="submitCaption" type="submit" color="primary" @click="onAddProvider" />
+      </q-card-actions>
+    </q-card>
+  </q-dialog>
+</template>
+
+<script lang="ts">
+export default defineComponent({
+  name: 'AddIdentityProviderDialog',
+});
+</script>
+<script setup lang="ts">
+import { IDProviderDto } from 'src/models/Opal';
+import { notifyError } from 'src/utils/notify';
+
+interface DialogProps {
+  modelValue: boolean;
+  provider: IDProviderDto | null;
+}
+
+const { t } = useI18n();
+const identityProvidersStore = useIdentityProvidersStore();
+const formRef = ref();
+const props = defineProps<DialogProps>();
+const emit = defineEmits(['update:modelValue']);
+const showDialog = ref(props.modelValue);
+const emptyProvider = {
+  name: '',
+  clientId: '',
+  secret: '',
+  discoveryURI: '',
+  scope: 'openid',
+  useNonce: true,
+  useLogout: true,
+  parameters: [],
+  enabled: false,
+  connectTimeout: 0,
+  readTimeout: 0,
+} as IDProviderDto;
+
+const groupsMappingOptionsIndices = {
+  groupsClaim: 0,
+  groupsScript: 1,
+};
+const groupsMappingOptions = ref([
+  { label: t('identity_provider.groups_claim'), value: 'groupsClaim' },
+  { label: t('identity_provider.groups_javascript'), value: 'groupsScript' },
+]);
+const groupsMapping = ref(groupsMappingOptions.value[groupsMappingOptionsIndices.groupsClaim]);
+const newProvider = ref<IDProviderDto>(emptyProvider);
+const editMode = computed(() => !!props.provider && !!props.provider.name);
+const submitCaption = computed(() => (editMode.value ? t('update') : t('add')));
+const dialogTitle = computed(() => (editMode.value ? t('identity_provider_edit') : t('identity_provider_add')));
+const isGroupsMappingByClaim = computed(() => groupsMapping.value.value === 'groupsClaim');
```

Comment:
If you use emit-value and map-options on q-select, it will simplify to `groupsMapping.value === 'groupsClaim'`

Reply:
Fixed